de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/auth/filter/CorsFilter.java

```diff
@@ -0,0 +1,78 @@
+/*******************************************************************************
+ * Copyright (c) 2020 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package de.dlr.sc.virsat.server.auth.filter;
+
+import java.io.IOException;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.container.ContainerResponseFilter;
+import javax.ws.rs.container.PreMatching;
+import javax.ws.rs.core.Response;
+
+import org.eclipse.jetty.http.HttpMethod;
+
+/**
+ * A filter that handles CORS requests
+ */
+@PreMatching
+public class CorsFilter implements ContainerRequestFilter, ContainerResponseFilter {
+
+	public static final String AC_ALLOW_CREDENTIALS = "Access-Control-Allow-Credentials";
+	public static final String AC_ALLOW_METHODS = "Access-Control-Allow-Methods";
+	public static final String AC_ALLOW_HEADERS = "Access-Control-Allow-Headers";
+	public static final String AC_ALLOW_ORIGIN = "Access-Control-Allow-Origin";
+	
+	public static final String AC_ALLOWED_CREDENTIALS = "true";
+	public static final String AC_ALLOWED_METHODS = "GET, POST, PUT, DELETE, OPTIONS, HEAD";
+	public static final String AC_ALLOWED_HEADERS = ""; //"X-Requested-With, Authorization, Accept-Version, Content-MD5, CSRF-Token, Content-Type";
```

Comment:
What is the purpose of the comment here?

Reply:
Ok so these are headers that we possible could/need to support at some point. I don't know if it's actually smart to leave them there or if we should delete them for now. I have no strong opinion on it.